index.js

```diff
@@ -30,12 +30,14 @@ function compile(options, getData) {
 }
 
 module.exports = function (data, options) {
-	return compile(options, function(file) {
-		if (file.data) {
-			data = _.merge(file.data, data);
-		}
-
-		return data || {};
+	return compile(options, function (file) {
+		return function (data) {
+			if (file.data) {
+				data = _.merge({}, file.data, data);
+			}
+
+			return data || {};
+		}(data);
```

Comment:
You’re right, it is much better this way :). I’ve updated the PR, do you want me to squash the last commit with the first?